core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -149,6 +241,53 @@ public SqlToConnectorExpressionTranslator(Map<NodeRef<Expression>, Type> types)
             return Optional.of(new Constant(null, typeOf(node)));
         }
 
+        @Override
+        protected Optional<ConnectorExpression> visitFunctionCall(FunctionCall node, Void context)
+        {
+            if (node.getFilter().isPresent() || node.getOrderBy().isPresent() || node.getWindow().isPresent() || node.getNullTreatment().isPresent() || node.isDistinct()) {
+                return Optional.empty();
+            }
+
+            String functionName = ResolvedFunction.extractFunctionName(node.getName());
+
+            if (LiteralFunction.LITERAL_FUNCTION_NAME.equalsIgnoreCase(functionName)) {
+                ExpressionInterpreter interpreter = new ExpressionInterpreter(node, metadata, session, types);
+                Object value = interpreter.evaluate();
+                if (value instanceof JoniRegexp) {
```

Comment:
This needs to work both for JoniRegexp and Re2Regexp

Reply:
Done